myapp/tests/controllers/UserController.test.js

```diff
@@ -0,0 +1,1412 @@
+const httpMocks = require('node-mocks-http');
+const UserController = require('../../controllers/UserController');
+const redisClient = require('../../databases/redisClient');
+const states = require('../../constants/states');
+const {
+  USER_HASH_PREFIX,
+  FOLLOWING_HASH_PREFIX,
+  FOLLOWER_HASH_PREFIX,
+  FRIENDS_HASH_PREFIX,
+  NEAR_FRIENDS_HASH_PREFIX,
+  NEAR_USERS_HASH_PREFIX,
+  ADDRESS_HASH_PREFIX,
+  USER_ID_SORTED_SET,
+  GEO_SORTED_SET,
+} = require('config').get('User');
+const User = require('../../models/User');
+const Address = require('../../models/Address');
+const UserRepository = require('../../repositories/UserRepository');
+const { EventEmitter } = require('events');
+
+const initDB = async (numberOfRecords) => {
+  const commands = [];
+  let user, userKey, addressKey;
+
+  for (let i = 1; i <= numberOfRecords; i++) {
+    userKey = `${USER_HASH_PREFIX}:${i}`;
+    addressKey = `${USER_HASH_PREFIX}:${i}:${ADDRESS_HASH_PREFIX}`;
+    user = new User({
+      id: i,
+      name: `test${i}`,
+      dob: '2021-02-28',
+      address: addressKey,
+      description: `Student ${i}`
+    });
+    const address = new Address({
+      longtitude: `50.00${i}`,
+      latitude: `-50.00${i}`,
+      description: `Address ${i}`
+    });
+    commands.push(
+      ['ZADD', USER_ID_SORTED_SET, i, userKey],
+      ['HSET', userKey, ...Object.entries(user).flat()],
+      ['HSET', addressKey, ...Object.entries(address).flat()],
+      ['GEOADD', GEO_SORTED_SET, address.longtitude, address.latitude, userKey]
+    );
+  }
+
+  commands.push(['SET', USER_HASH_PREFIX, numberOfRecords]);
+  await redisClient.batch(commands);
+};
+
+const cleanDB = async () => {
```

Comment:
Perhaps use `FLUSHALL` for `cleanDB` could be more reliable and easy to maintain.

Reply:
Whether to use FLUSHALL depends on the following cases:

* If no other developer is performing testing against the configured redis server, it is a good choice to use FLUSHALL command.
* If the redis server is shared by multiple people who could be performing their testing at the same time, I don't believe FLUSHALL command is a better idea.
* Another good practice is to provide value for the prefix parameter when creating the redis client. So that for each command it will append this prefix to the specified key before sending each command. So 'namespace:test' could be used as a prefix when executing those test cases.